### PR TITLE
Blog app - controllers

### PR DESCRIPTION
**Pull request by VaheStepanyan100**
-  **Important:** do NOT use `rails generate scaffold` command. The point of this project is to show how to set up controllers by yourself.
- Make sure that app handles the URLs used in the wireframes: 
  - All users page wireframe
  - Single user page wireframe
  - All posts by a given user page wireframe
  - Single post page wireframe

-  For each URL should be created:
    - A route.
    - An action in the correct controller.
    - A view file.

-  View files should include only placeholders written in plain HTML, e.g:
    - Here is a list of posts for a given user

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    if @post.save
+      flash[:notice] = 'Post created successfully.'
+      redirect_to user_path(current_user)
+    else
+      render 'new'
+    end
```
Comment (Davidosky007) [inline app/controllers/posts_controller.rb]: * As stated in the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md#:~:text=NOT%20use%20rails%20generate%20scaffold%20command.%20The%20point%20of%20this%20project%20is%20to%20show%20you%20how%20to%20set%20up%20controllers%20by%20yourself.) NOT to use rails generate scaffold command. The point of this project is to show you how to set up controllers by yourself. Only create the contorllers and actions that you need For example 👇 
```
# app/controllers/users_controller.rb
class UsersController < ApplicationController
  
  def index; end

  def show; end

end

# app/controllers/posts_controller.rb
class PostsController < ApplicationController
  
  def index; end

  def show; end

end
```

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    @users = User.all
+  end
+
+  def show; end
+
+  def new; end
+
+  def find_user
+    @user = User.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
```
Comment (Davidosky007) [inline app/controllers/users_controller.rb]: * As stated in the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md#:~:text=NOT%20use%20rails%20generate%20scaffold%20command.%20The%20point%20of%20this%20project%20is%20to%20show%20you%20how%20to%20set%20up%20controllers%20by%20yourself.) NOT to use rails generate scaffold command. The point of this project is to show you how to set up controllers by yourself. Only create the contorllers and actions that you need For example 👇 
```
# app/controllers/users_controller.rb
class UsersController < ApplicationController
  
  def index; end

  def show; end

end

# app/controllers/posts_controller.rb
class PostsController < ApplicationController
  
  def index; end

  def show; end

end
```

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<h1><%= @user.name%>'s Posts</h1>
+<ul>
+<% @posts.each do |post|%>
+<li>post_id: <%= post.id %>
+<span>post_title: <%= post.title %></span>
+<% end %>
```
Comment (Davidosky007) [inline app/views/posts/index.html.erb]: * As stated in the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md#:~:text=View%20files%20should,a%20given%20user) View files should include only placeholders written in plain HTML, e.g Here is a list of posts for a given user

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<h1>Post number <%= @post.id %> of User <%= @user.name %></h1>
```
Comment (Davidosky007) [inline app/views/posts/show.html.erb]: * As stated in the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md#:~:text=View%20files%20should,a%20given%20user) View files should include only placeholders written in plain HTML, e.g Here is a list of posts for a given user

File: app/views/users/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<h1>Users</h1>
```
Comment (Davidosky007) [inline app/views/users/index.html.erb]: * As stated in the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md#:~:text=View%20files%20should,a%20given%20user) View files should include only placeholders written in plain HTML, e.g Here is a list of posts for a given user

File: app/views/users/show.html.erb [hunk 6]
```diff
@@ -0,0 +1 @@
+<h1><%= @user.name%></h1>
```
Comment (Davidosky007) [inline app/views/users/show.html.erb]: * As stated in the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md#:~:text=View%20files%20should,a%20given%20user) View files should include only placeholders written in plain HTML, e.g Here is a list of posts for a given user

File: config/routes.rb [hunk 7]
```diff
@@ -5,4 +5,7 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  resources :users do
+    resources :posts
+  end
```
Comment (Davidosky007) [inline config/routes.rb]: * As stated in the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md#:~:text=Make%20sure%20that%20you%20added,can%20see%20the%20following%20output%3A) to Make sure that you add routes to handle only the 4 required URLs.
* currently all the unnecessary endpoints as being exposed, Kindly limit your routes to only actions needed in your controllers For example 👇 

```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @VaheStepanyan100 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well written PR 👍 
* No linter errors 👍 
* Good looking readme 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @VaheStepanyan100,

✔️ Implementation of nested routes

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**